core/src/kvs/ds.rs

```diff
@@ -376,6 +381,22 @@ impl Datastore {
 				}
 				#[cfg(not(feature = "kv-fdb"))]
                 return Err(Error::Ds("Cannot connect to the `foundationdb` storage engine as it is not enabled in this build of SurrealDB".to_owned()));
+			}
+			// Parse and initiate a SurrelKV database
+			s if s.starts_with("surrealkv:") => {
```

Comment:
Won't this be replacing `file://`? We could have it as a conditional build on file. But happy with the current solution as well

Reply:
I'm not really sure about it, should it replace?

Reply:
Agree with Hugh. At term, `file://...` should be SurrealKV.
RocksDB could then have a "vendor" path: `rockdb://...`
May be for 2.0 (as it would be a breaking change)?
Or, giving up with `file://...` and have a vendor orientied logic?
`rocksdb://...` and `skv://...`

Reply:
We shouldn't replace `file://` just yet, for a couple of reasons:-

1. SurrealKV is still experimental.
2. Doing it in the middle of a stable release would be a breaking change. If SurrealKV stabilises before v2 then we can switch it then. If not, it will have to wait for v3 before we can make that switch.

Reply:
Yes, we shouldn't replace it. But if we build it with a feature flag then it could replace. Like speedb does

Reply:
SpeeDB doesn't replace either @phughk, it uses the `speedb` scheme to select the SpeeDB engine. RocksDB also has a specific `rocksdb` scheme too. `file` is currently an alias for `rocksdb`. So we should add the specific `surrealkv` scheme for now and leave `file` pointing to `rocksdb`. Once SurrealKV is now stable and we are moving to a new major version we can make `file` point to `surrealkv`.

Reply:
I'll keep it the same as it is for now, and once stable, we can look to change this